### PR TITLE
ISPN-14362 Json escape should use two equality checks instead of Hash…

### PR DESCRIPTION
**Pull request by jabolina**
…Map lookup

https://issues.redhat.com/browse/ISPN-14362

And removes the unused HTML escape.

### CODE REVIEW

File: commons/all/src/main/java/org/infinispan/commons/dataconversion/internal/Json.java [hunk 1]
```diff
@@ -2891,8 +2867,7 @@ private void escapeJsonString(CharSequence plainText, Appendable out) throws IOE
       private boolean mustEscapeCharInJsString(int codepoint) {
          if (!Character.isSupplementaryCodePoint(codepoint)) {
```
Comment (wburns) [inline commons/all/src/main/java/org/infinispan/commons/dataconversion/internal/Json.java:2868]: I am questioning this line here as well now.. I am thinking we can just replace this all with the `return c == '"' || c == '\\';` line

Comment (jabolina) [reply commons/all/src/main/java/org/infinispan/commons/dataconversion/internal/Json.java:2868]: It seems that is safe to do. I'll update and we see CI.

Comment (jabolina) [reply commons/all/src/main/java/org/infinispan/commons/dataconversion/internal/Json.java]: CI seems ok.

**Comment by jabolina** [comment]
Some numbers from the benchmark:

```text
Benchmark                       (stringSize)  (type)   Mode  Cnt       Score      Error  Units
JsonBenchmark.testStringEscape             1     new  thrpt   25  330501.497 ± 2221.774  ops/s
JsonBenchmark.testStringEscape             1    main  thrpt   25  178204.141 ± 3869.935  ops/s
JsonBenchmark.testStringEscape          1024     new  thrpt   25     345.199 ±    4.689  ops/s
JsonBenchmark.testStringEscape          1024    main  thrpt   25     161.054 ±    1.660  ops/s
```

**Comment by tristantarrant** [comment]
Please create a backport for 14.0.x

**Pull request merged**